### PR TITLE
chore: fix generate contributors action

### PR DESCRIPTION
**Pull request by armano2**
## PR Checklist

-   [ ] Addresses an existing open issue: fixes #000
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

replace all-contributors cli with small script and fix execution of `Generate contributors` action https://github.com/typescript-eslint/typescript-eslint/runs/6243134418?check_suite_focus=true

### CODE REVIEW

File: tools/generate-contributors.ts [hunk 1]
```diff
@@ -78,40 +133,16 @@ async function main(): Promise<void> {
...
   const users = await Promise.all(
-    githubContributors.map(async c => {
-      const response = await fetch(c.url, { method: 'GET' });
-      return (await response.json()) as User;
-    }),
+    githubContributors
+      // remove ignored users and bots
+      .filter(
+        usr => usr.login && usr.type !== 'Bot' && !IGNORED_USERS.has(usr.login),
+      )
```
Comment (armano2) [inline tools/generate-contributors.ts:140]: we actually don't have to drop this contributors cli, all what has to be done is this change as some users may not have their profiles set to public, 

with latest github api we are calling those endpoints only to retrieve "name", as all other data is available already to us from contributions api

**Comment by armano2** [comment]
looks like cypress failed again

File: tools/generate-contributors.ts [hunk 2]
```diff
@@ -68,6 +73,56 @@ async function* fetchUsers(page = 1): AsyncIterableIterator<Contributor[]> {
...
+    lines.push('  </tr>');
+  }
+
+  lines.push('</table>');
+  lines.push('');
+  lines.push('<!-- markdownlint-restore -->');
+  lines.push('<!-- prettier-ignore-end -->');
+  lines.push('');
+  lines.push(
+    "<sup>'This list is auto-generated using `yarn generate-contributors`. It shows the top 100 contributors with > 3 contributions.'</sup>",
```
Comment (bradzacher) [inline tools/generate-contributors.ts:116]:
```suggestion
    `<sup>This list is auto-generated using `yarn generate-contributors`. It shows the top 100 contributors with > ${COMPLETELY_ARBITRARY_CONTRIBUTION_COUNT} contributions.</sup>`,
```

File: tools/generate-contributors.ts (hunk 1 above)
Comment (bradzacher) [reply tools/generate-contributors.ts:140]: more than happy to remove a dependency here!

**Review by bradzacher — DISMISSED** [review]
This LGTM! Happy to remove a clunky dependency!

File: tools/generate-contributors.ts (hunk 2 above)
Comment (armano2) [reply tools/generate-contributors.ts:116]:
```suggestion
    `<sup>This list is auto-generated using \`yarn generate-contributors\`. It shows the top ${PAGE_LIMIT} contributors with > ${COMPLETELY_ARBITRARY_CONTRIBUTION_COUNT} contributions.</sup>`,
```

**Review by bradzacher — DISMISSED** [review]

**Comment by armano2** [comment]
looks like auto-merge failed due to cypress

**Comment by JoshuaKGoldberg** [comment]
_shakes fist at Cypress_

**Pull request merged**